### ISSUE TITLE
Access to Harmony Devices

### ISSUE THREAD

**Issue by apwelsh** [ogiewon/Hubitat#8]
Is there any way to get this to return devices too, so that we can directly turn on/off devices without an activity?  I would like to be able to turn off my TV even if turned on via another source.

**Comment by ogiewon** [comment]
The webSockets API that is used by this integration actually does expose the individual devices.  I just personally have not had a need to control each device.  Activities are enough for my needs.  If you would like to extend the capabilities of this integration to include individual devices, I would welcome a pull request once your changes are fully tested.

**Comment by apwelsh** [comment]
I will research. Thanks 

Armand
--Sent from iPhone

**Comment by apwelsh** [comment]
I found the API documentation for the websocket solution, and am working on the solution.  Not sure how to implement w/o breaking the current system.  I need a way to track that the Child Switch maps to a Device or Activity.  Thought about using state in the parent, but then it would lose sync with child switches.  How do you feel about adding state to the child switch? (not sure if it is possible yet)

**Comment by ogiewon** [comment]
I believe the simplest way to know whether a child is a “device” versus and “activity” is to store that information as part of the child’s Device Network ID.

**Comment by ogiewon** [comment]
Any progress on this?  If not, I will close the issue.  Thanks!

**Comment by apwelsh** [comment]
The Harmony does not capture state and does not attempt to.  I was able to send basic commands like power-on and power-off, and I used a switch child handler to manage state, but if you used the harmony, there was no way to no that the device was on or off to keep the switch in sync. As such, I gave up on trying to make the harmony intelligent enough to work for my needs, and switched over to creating a device handler for my Roku TV that works with all my Roku devices, and does maintain state by proving the device for state, and because I talk to the TV, I implemented a remote interface that does everything as well as enabling various child devices based on toggles. 
I still use this to control my IR devices, but the Harmony just is not powerful enough to rely on for network enabled devices (which is what I was using it for)
